integration/container/rename_test.go

```diff
@@ -86,3 +91,195 @@ func TestRenameLinkedContainer(t *testing.T) {
+	assert.Equal(t, inspect.Name, "/"+newName)
+}
+
+func TestRenameRunningContainerAndReuse(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	oldName := "first_name"
+	cID := runSimpleContainer(ctx, t, client, oldName)
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err := client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
+	err = client.ContainerRename(ctx, oldName, newName)
+	require.NoError(t, err)
+
+	inspect, err = client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+newName)
+
+	cID = runSimpleContainer(ctx, t, client, oldName)
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err = client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+}
+
+func TestRenameCheckNames(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	oldName := "first_name"
+	cID := runSimpleContainer(ctx, t, client, oldName)
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err := client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+
+	newName := "new_name" + stringid.GenerateNonCryptoID()
+	err = client.ContainerRename(ctx, oldName, newName)
+	require.NoError(t, err)
+
+	_, err = client.ContainerInspect(ctx, oldName)
+	testutil.ErrorContains(t, err, "No such container: "+oldName)
+}
+
+func TestRenameInvalidName(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	oldName := "first_name"
+	cID := runSimpleContainer(ctx, t, client, oldName)
+	poll.WaitOn(t, containerIsInState(ctx, client, cID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	inspect, err := client.ContainerInspect(ctx, cID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.Name, "/"+oldName)
+
+	err = client.ContainerRename(ctx, oldName, "new:invalid")
+	testutil.ErrorContains(t, err, "Invalid container name")
+
+	inspect, err = client.ContainerInspect(ctx, oldName)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.ID, cID)
+}
+
+func TestRenameAnonymousContainer(t *testing.T) {
+	defer setupTest(t)()
+	ctx := context.Background()
+	client := request.NewAPIClient(t)
+
+	_, err := client.NetworkCreate(ctx, "network1", types.NetworkCreate{})
+	require.NoError(t, err)
+	cID := createSimpleContainer(ctx, t, client, "", func(config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig) {
+		networkingConfig.EndpointsConfig = map[string]*network.EndpointSettings{
+			"network1": {},
+		}
+		hostConfig.NetworkMode = "network1"
```

Comment:
What does this network stuff have to do with the container being anonymous? I think the test name needs to include more context or a godoc comment needs to be added to the test function explaining the case.

I understand you are just porting this test, but if we don't make sense of these tests now then we're just porting debt.

Reply:
Added the description of the test with link to GitHub issue. Thanks.